src/metadata.rs

```diff
@@ -153,6 +153,25 @@ impl Metadata21 {
                 }
             }
 
+            // Until PEP 639 is approved with metadata 2.3, we can assume a
+            // dynamic license-files (also awaiting full 2.2 metadata support)
+            // We're already emitting the License-Files metadata without issue.
+            // license-files.globs = ["LICEN[CS]E*", "COPYING*", "NOTICE*", "AUTHORS*"]
+            let license_include_targets = ["LICEN[CS]E*", "COPYING*", "NOTICE*", "AUTHORS*"];
+            for pattern in license_include_targets.iter() {
+                for license_path in glob::glob(&manifest_path.join(pattern).to_string_lossy())
+                    .expect("No license files found for pattern")
```

Comment:
the `manifest_path` needs to be escaped with `glob::escape` or it'll become part of the pattern. The expect shouldn't happen anyway, but a`?` would be better than possibly panicking

Reply:
I pushed a fix for the `?` usage instead of the `expect`. I followed the pattern from [source_distribution.rs#L317-L328](https://github.com/PyO3/maturin/blob/main/src/source_distribution.rs#L317-L328). I can update the code to used `Pattern::escape` if you like.

```rust
    if let Some(include_targets) = sdist_include {
        for pattern in include_targets {
            println!("📦 Including files matching \"{}\"", pattern);
            for source in glob::glob(&manifest_dir.join(pattern).to_string_lossy())
                .expect("No files found for pattern")
                .filter_map(Result::ok)
            {
                let target = root_dir.join(&source.strip_prefix(manifest_dir)?);
                writer.add_file(target, source)?;
            }
        }
    }
```

Reply:
@konstin Is this what you are looking for?
```rust
let license_include_targets = ["LICEN[CS]E*", "COPYING*", "NOTICE*", "AUTHORS*"];
let escaped_manifest_string = glob::Pattern::escape(manifest_path.to_str().unwrap());
let escaped_manifest_path = Path::new(&escaped_manifest_string);
for pattern in license_include_targets.iter() {
    for license_path in
        glob::glob(&escaped_manifest_path.join(pattern).to_string_lossy())?
            .filter_map(Result::ok)
    {
//....
```

Reply:
yep that looks good!

Reply:
@konstin Just pushed the change. Thanks for the tip.